### DISCUSSION TITLE
Integrating golang ci lint report in Jenkins

### DISCUSSION THREAD

**Discussion by arthurvaverko** [golangci/golangci-lint#4442] [Q&A] [answered]
We are using jenkins for our CI .. i was wondering what report tools do you use to view the lint errors and warnings in Jenkins ? 
I'm looking for something i can place a gate and fail the build in case some lint errors occur ..

**Comment by bombsimon** [comment, marked as answer]
You can use the [JUnit plugin](https://plugins.jenkins.io/junit/) for Jenkins and then use `junit-xml` as out format for `golangci-lint`.

And by just googling now I found [`jenkinsci/warnings-ng-plugin`](https://github.com/jenkinsci/warnings-ng-plugin) that supports checkstyle which `golangci-lint` will report if you set out format to `checkstyle`.
